source/math/FloatVectorList.ooc

```diff
@@ -247,15 +235,93 @@ FloatVectorList: class extends VectorList<Float> {
 		tempVector free()
 		result
 	}
+	/* calculate median without copying and sorting the vector
+			WARNING: this function can partially rearange the elements in vector
+	*/
+	fastMedian: func -> Float {
+		data := this _vector _backend as Float*
+		result := This _nthElement(this _vector _backend as Float*, 0, this count - 1, this count / 2)
+		if (Int even(this count))
+			result = (result + This _nthElement(this _vector _backend as Float*, 0, this count - 1, this count / 2 - 1)) / 2
+		result
+	}
 	movingMedianFilter: func (windowSize: Int) -> This {
 		result := This new(this count)
 		start := -((windowSize - 1) / 2)
 		for (i in 0..this count) {
 			range := (start..(start + windowSize - 1)) + i
 			elementsInWindow := this getSlice(range clamp(0, this count-1))
-			result add((elementsInWindow as This) median())
+			result add((elementsInWindow as This) fastMedian())
 			elementsInWindow free()
 		}
 		result
 	}
+	_swap: static func (array: Float*, i, j: Int) {
+		t := array[i]
+		array[i] = array[j]
+		array[j] = t
+	}
+	/* partition the array so that all elements less than array[pivot] are moved before this element
+		and all elements greater than array[pivot] are after it
+		end should be the last valid array index in the range
+	*/
+	_partition: static func (array: Float*, start, end, pivot: Int) -> Int {
+		pivotValue := array[pivot]
+		This _swap(array, pivot, end)
+		result := start
+		for (i in start..end)
+			if (array[i] < pivotValue) {
+				This _swap(array, result, i)
+				++result
```

Comment:
To me it feels so natural to use it that I just used it, without even thinking about it. I just assumed that it exsits in ooc, because, well, why it shouldn't :)